userspace/falco/falco.cpp

```diff
@@ -564,257 +450,49 @@ int falco_init(int argc, char **argv)
-				break;
-			case 'V':
-				validate_rules_filenames.push_back(optarg);
-				break;
-			case 'w':
-				outfile = optarg;
-				break;
-			case '?':
-				result = EXIT_FAILURE;
-				goto exit;
+		std::string errstr;
+		bool successful = app.init(argc, argv, errstr);
 
-			case 0:
-				if(string(long_options[long_index].name) == "version")
-				{
-					printf("Falco version: %s\n", FALCO_VERSION);
-					printf("Driver version: %s\n", DRIVER_VERSION);
-					return EXIT_SUCCESS;
-				}
-				else if (string(long_options[long_index].name) == "cri")
-				{
-					if(optarg != NULL)
-					{
-						cri_socket_path = optarg;
-					}
-				}
-				else if (string(long_options[long_index].name) == "disable-cri-async")
-				{
-				  cri_async = false;
-				}
-#ifndef MINIMAL_BUILD
-				else if(string(long_options[long_index].name) == "k8s-node")
-				{
-					k8s_node_name = new string(optarg);
-					if (k8s_node_name->size() == 0) {
-						throw std::invalid_argument("If --k8s-node is provided, it cannot be an empty string");
-					}
-				}
-#endif
-				else if (string(long_options[long_index].name) == "list")
-				{
-					list_flds = true;
-					if(optarg != NULL)
-					{
-						list_flds_source = optarg;
-					}
-				}
-#ifndef MUSL_OPTIMIZED
-				else if (string(long_options[long_index].name) == "list-plugins")
-				{
-					list_plugins = true;
-				}
-#endif
-				else if (string(long_options[long_index].name) == "stats-interval")
-				{
-					stats_interval = atoi(optarg);
-				}
-				else if (string(long_options[long_index].name) == "support")
-				{
-					print_support = true;
-				}
-				else if (string(long_options[long_index].name) == "disable-source")
-				{
-					if(optarg != NULL)
-					{
-						disable_sources.insert(optarg);
-					}
-				}
-				break;
+		if(!successful)
+		{
+			throw falco_exception(string("Could not initialize: ") + errstr);
+		}
 
-			default:
-				break;
-			}
+		if(app.copts().help)
+		{
+			printf("%s", app.copts().usage().c_str());
+			return EXIT_SUCCESS;
+		}
 
+		if(app.copts().print_version_info)
+		{
+			printf("Falco version: %s\n", FALCO_VERSION);
+			printf("Driver version: %s\n", DRIVER_VERSION);
+			return EXIT_SUCCESS;
 		}
 
 		inspector = new sinsp();
-		inspector->set_buffer_format(event_buffer_format);
+		inspector->set_buffer_format(app.copts().event_buffer_format);
 
 		// If required, set the CRI path
-		if(!cri_socket_path.empty())
+		if(!app.copts().cri_socket_path.empty())
 		{
-			inspector->set_cri_socket_path(cri_socket_path);
+			inspector->set_cri_socket_path(app.copts().cri_socket_path);
 		}
 
 		// Decide wether to do sync or async for CRI metadata fetch
-		inspector->set_cri_async(cri_async);
+		inspector->set_cri_async(!app.copts().disable_cri_async);
 
 		//
 		// If required, set the snaplen
 		//
-		if(snaplen != 0)
+		if(app.copts().snaplen != 0)
 		{
-			inspector->set_snaplen(snaplen);
+			inspector->set_snaplen(app.copts().snaplen);
 		}
 
 		if(print_ignored_events)
```

Comment:
Where is this initialized, and why is it not part of `app.copts()`?

Reply:
Most of the options have a `->default_value(xxx)` call which sets the default value for the option. Exceptions are for options that are bound to strings that can't be used without the option being specified (e.g. --cri <path> to specify an alternate cri socket, --disable-source to disable a source), and --list_plugins, which may be commented out for MUSL_OPTIMIZED builds.

The default for snaplen is 0 (https://github.com/falcosecurity/falco/blob/add-cmdline-opts-library/userspace/falco/app_cmdline_options.cpp#L181) which matches what it was set to for falco before this PR (https://github.com/falcosecurity/falco/blob/8d9dd4440fa9e28911342d6e668f62062b263fff/userspace/falco/falco.cpp#L532)

Reply:
I'm not sure I'm understanding well, how this is related to `print_ignored_events`?

Reply:
Oh sorry--I thought you meant snaplen. I missed this, it used to be the `-i` option. I'll add it.